### PR TITLE
React sidebar

### PR DESCRIPTION
**Pull request by garrensmith**
This converts the sidebar to react

### CODE REVIEW

File: app/addons/components/tests/codeEditorSpec.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  var code2 = 'function (doc) {\n if(doc._id) { \n emit(doc._id, 2); \n } \n}';
+
+  describe('Code Editor', function () {
+    var container, codeEditorEl, spy;
+
+    beforeEach(function () {
+      spy = sinon.spy();
+      container = document.createElement('div');
+      codeEditorEl = TestUtils.renderIntoDocument(
+        React.createElement(ReactComponents.CodeEditor, {code: code, change: spy}),
```
Comment (robertkowalski) [inline app/addons/components/tests/codeEditorSpec.js]: that is checked in code from react

File: app/addons/documents/sidebar/sidebar.react.jsx [hunk 2]
```diff
@@ -0,0 +1,461 @@
...
+        $dashboard.animate({left: '210px'}, 300);
+      } else {
+        $dashboard.animate({left: '550px'}, 300);
+      }
+
+    },
+
+    render: function () {
+      return null;
+      /*var text = this.state.hidden ? 'show sidebar' : 'hide';
```
Comment (robertkowalski) [inline app/addons/documents/sidebar/sidebar.react.jsx]: ?

File: app/addons/documents/sidebar/stores.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    },
+
+    isVisible: function (designDoc, index) {
+      if (!this._toggledSections[designDoc]) {
+        return false;
+      }
+
+      if (index) {
+        var x = this._toggledSections[designDoc].indexes[index];
+        console.log('in', index, x, designDoc);
```
Comment (robertkowalski) [inline app/addons/documents/sidebar/stores.js]: debug output

File: app/addons/documents/sidebar/tests/sidebar.actionsSpec.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  var assert = testUtils.assert;
+  var restore = testUtils.restore;
+  var store = Stores.sidebarStore;
+
+  FauxtonAPI.router = new FauxtonAPI.Router([]);
+
+  describe('Sidebar Actions', function () {
+
+  });
+
```
Comment (robertkowalski) [inline app/addons/documents/sidebar/tests/sidebar.actionsSpec.js]: this file is empty

File: app/addons/documents/sidebar/tests/sidebar.componentsSpec.react.jsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  var assert = utils.assert;
+  var TestUtils = React.addons.TestUtils;
+
+  describe('Sidebar Components', function () {
+    var container, el;
+
+    beforeEach(function () {
+      container = document.createElement('div');
+    });
+
```
Comment (robertkowalski) [inline app/addons/documents/sidebar/tests/sidebar.componentsSpec.react.jsx]: there are no tests

File: assets/less/templates.less [hunk 6]
```diff
@@ -453,8 +453,8 @@ with_tabs_sidebar.html
     .divider {
       border: none;
     }
-    > li > a:hover,
-    > li > a:focus {
+    > li > a:hover{
+    //> li > a:focus {
```
Comment (robertkowalski) [inline assets/less/templates.less]: do we still need that?

File: app/addons/components/react-components.react.jsx [hunk 7]
```diff
@@ -445,6 +445,62 @@ function (app, FauxtonAPI, React, Components, ace, beautifyHelper) {
     }
   });
 
+  var MenuDropDown = React.createClass({
+
+    getDefaultProps: function () {
+      return {
+        icon: "fonticon-plus-circled"
```
Comment (benkeen) [inline app/addons/components/react-components.react.jsx]: single quote

File: app/addons/documents/sidebar/stores.js [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+    },
+
+    isVisible: function (designDoc, index) {
+      if (!this._toggledSections[designDoc]) {
+        return false;
+      }
+
+      if (index) {
+        var x = this._toggledSections[designDoc].indexes[index];
+        return x;
```
Comment (benkeen) [inline app/addons/documents/sidebar/stores.js]: Here we could just return it rather than stash it in a var first.

**Comment by benkeen** [comment]
+1 Looks awesome. Code + functionality looks great - just a couple of very minor things I mentioned above.

**Comment by michellephung** [comment]
merged! 9d7cd44213a56a0a84486048c105b5c9c62c04ee

**Comment by benkeen** [comment]
Nice! This can be closed now.

**Pull request closed without merging**